tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/sr_evaluator.py

```diff
@@ -0,0 +1,572 @@
+        self._name_to_idx = {}
+        self._feedback_name = self.network_info['feedback_input']
+        self._feedback_data = {self._feedback_name: None}
+        self._first_step = True
+        self._inputs = self.network_info['inputs']
+        self._feedback_inputs = {self._feedback_name: [t for t in self._inputs if t['name']==self._feedback_name][0]}
+
+        for input_info in self._inputs:
+            idx = int(input_info['value'])
+            self._idx_to_name[idx] = input_info['name']
+            self._name_to_idx[input_info['name']] = idx
+        self._feedback_idx = self._name_to_idx[self._feedback_name]
+
+
+    def init_feedback(self, reader):
+        info = self._feedback_inputs[self._feedback_name]
+        self._feedback_data[self._feedback_name] = reader.read(info['initializer'])
+
+    def feedback(self, data):
+        data = data[0]
+        # info = self._feedback_inputs[self._feedback_name]
+        self._feedback_data[self._feedback_name] = data[0].value
+
+    def fill_feedback(self, data):
+        data[0].data[self._feedback_idx] = self._feedback_data[self._feedback_name]
+        return data
+
+
+class ModelDLSDKModel(BaseModel, BaseDLSDKModel, FeedbackMixin):
+    default_model_suffix = 'model'
+
+    def __init__(self, network_info, launcher, delayed_model_loading=False):
+        super().__init__(network_info, launcher)
+        self.input_blob, self.output_blob = None, None
+        self.with_prefix = None
+        if not delayed_model_loading:
+            self.load_model(network_info, launcher, log=True)
+
+        self.adapter = create_adapter(network_info.get('adapter', 'super_resolution'))
+        self.configure_feedback()
+
+    def predict(self, identifiers, input_data):
+        input_data = self.fit_to_input(input_data)
+        raw_result = self.exec_network.infer(input_data)
+        result = self.adapter.process([raw_result], identifiers, [{}])
+        return raw_result, result
+
+    def release(self):
+        del self.exec_network
+        del self.launcher
+
+    def fit_to_input(self, input_data):
+        has_info = hasattr(self.exec_network, 'input_info')
+        if has_info:
+            input_info = self.exec_network.input_info
+        else:
+            input_info = self.exec_network.inputs
+
+        fitted = {}
+        for name, info in input_info.items():
+            data = input_data[self._name_to_idx[name]]
+            data = np.expand_dims(data, axis=0)
+            data = np.transpose(data, [0, 3, 1, 2])
+            assert tuple(info.input_data.shape) == np.shape(data)
+            fitted[name] = data
+
+        return fitted
+
+
+class ModelTFModel(BaseModel, FeedbackMixin):
+    default_model_suffix = 'model'
+
+    def __init__(self, network_info, launcher, *args, **kwargs):
+        super().__init__(network_info, launcher)
+        model = self.automatic_model_search(network_info)
+        self.inference_session = launcher.create_inference_session(str(model))
```

Comment:
hmm... looks suspicious, do you tests tf model running? this code part more simmilar with onnx runtime than tf

Reply:
Yes, TF test was passed successfully